app/controllers/concerns/discovery.rb

```diff
@@ -1,18 +1,47 @@
+require "velum/salt_minion"
+require "velum/salt_api"
 # Discovery implements the discovery method that is shared both when
 # bootstrapping and when showing the dashboard.
 module Discovery
   extend ActiveSupport::Concern
+  include Velum::SaltApi
 
   # Responds with either an HTML or JSON version of the available minions.
   def discovery
     @assigned_minions = Minion.assigned_role
     @unassigned_minions = Minion.unassigned_role
 
+    tx_update_reboot_needed = Hash.new
+    tx_update_failed = Hash.new
+
+    minions = perform_request(endpoint: "/minions", method: "get")
```

Comment:
I am not sure about calling all the time here to the `salt-api`. This will the request coming from the user slower. We could add a cron job that updates the status of the minions on the background and we would only ask for this column on the database (since we are missing a resque-similar tool for background jobs, that's why a cron would be the best solution IMHO).

However this can be improved at a later time, is not that important.